storage_service: avoid processing same table unnecessarily in split monitor

raphaelsc opened the pull request:
If there's a token metadata for a given table, and it is in split mode, it will be registered such that split monitor can look at it, for example, to start split work, or do nothing if table completed it.

during topology change, e.g. drain, split is stalled since it cannot take over the state machine.
It was noticed that the log is being spammed with a message saying the table completed split work, since every tablet metadata update, means waking up the monitor on behalf of a table. So it makes sense to demote the logging level to debug. That persists until drain completes and split can finally complete.

Another thing that was noticed is that during drain, a table can be submitted for processing faster than the monitor can handle, so the candidate queue may end up with multiple duplicated entries for same table, which means unnecessary work. That is fixed by using a sequenced set, which keeps the current FIFO behavior.

Fixes #20339.

backport reason: avoids unneeded work in split monitor if split runs concurrently to topology changes. also avoids spamming the logs, in that scenario.

raphaelsc commented:
@tgrabiec please take a look.

raphaelsc commented:
> #### 🔴 CI State: ABORTED
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2922/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10944/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/)
> 
> #### Failed Tests (96/12609):
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.1.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> 
> #### Build Details:
> * Duration: 20 hr
> * Builder: spider8.cloudius-systems.com

@benipeled infra issue?

mykaul commented:
@raphaelsc - Beni is no longer working at Scylla. @yaronkaikov may be able to assist.

raphaelsc commented:
> @raphaelsc - Beni is no longer working at Scylla. @yaronkaikov may be able to assist.

oops. thanks for the update.

scylladb-promoter commented:
#### :red_circle: CI State: ABORTED

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10788/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10788/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10953/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2931/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10788/)
#### Failed Tests (64/12592):
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.1.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.1.xunit/xml/Tests___Unit_Tests____empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.2.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10788/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.2.xunit/xml/Tests___Unit_Tests____empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
#### Build Details:
- Duration: 11 hr
- Builder: spider2.cloudius-systems.com

yaronkaikov commented:
> > #### 🔴 CI State: ABORTED
> > ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2922/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10944/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/)
> > #### Failed Tests (96/12609):
> > 
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.1.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > 
> > #### Build Details:
> > 
> > * Duration: 20 hr
> > * Builder: spider8.cloudius-systems.com
> 
> @benipeled infra issue?

@raphaelsc I am not sure it's related, will try to search in the logs

yaronkaikov commented:
> > > #### 🔴 CI State: ABORTED
> > > ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2922/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10944/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/)
> > > #### Failed Tests (96/12609):
> > > 
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.1.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > 
> > > #### Build Details:
> > > 
> > > * Duration: 20 hr
> > > * Builder: spider8.cloudius-systems.com
> > 
> > 
> > @benipeled infra issue?
> 
> @raphaelsc I am not sure it's related, will try to search in the logs

It looks like one of the unit-tests was stack. but i can't find anything in the logs

yaronkaikov commented:
> > > > #### 🔴 CI State: ABORTED
> > > > ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2922/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10944/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10778/)
> > > > #### Failed Tests (96/12609):
> > > > 
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > * [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10778/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.1.xunit/xml/Tests___Unit_Tests____empty_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%5Bempty%5D)
> > > > 
> > > > #### Build Details:
> > > > 
> > > > * Duration: 20 hr
> > > > * Builder: spider8.cloudius-systems.com
> > > 
> > > 
> > > @benipeled infra issue?
> > 
> > 
> > @raphaelsc I am not sure it's related, will try to search in the logs
> 
> It looks like one of the unit-tests was stack. but i can't find anything in the logs

Started the CI again, https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10793/

yaronkaikov commented:
@raphaelsc it seems that the unit-tests stack again, maybe related to the code change?

raphaelsc commented:
> @raphaelsc it seems that the unit-tests stack again, maybe related to the code change?

I am not sure. full logs show it's stuck in boost unit tests, which don't stress code path changed at all.

tgrabiec commented on service/storage_service.cc:
```diff
@@ -5250,7 +5250,7 @@ future<> storage_service::load_tablet_metadata() {
 }
 
-future<> storage_service::process_tablet_split_candidate(table_id table) {
+future<> storage_service::process_tablet_split_candidate(table_id table) noexcept {
```
Why noexcept?

raphaelsc replied:
monitor expects this to not throw, so it will not be aborted. this is only about documenting it, since in practice, it shouldn't throw.

raphaelsc replied:
shouldn't throw = will not propagate exception

scylladb-promoter commented:
#### :red_circle: CI State: ABORTED

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10793/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10793/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2935/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10957/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10793/)
#### Failed Tests (96/12636):
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.1.xunit/xml/Tests___Unit_Tests____empty__3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.2.xunit/xml/Tests___Unit_Tests____empty__3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanup_record_gc.3.xunit/xml/Tests___Unit_Tests____empty__3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10793/testReport/junit/boost.commitlog_cleanup_test.test_commitlog_cleanups.1.xunit/xml/Tests___Unit_Tests____empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
#### Build Details:
- Duration: 20 hr
- Builder: spider2.cloudius-systems.com

raphaelsc commented:
I know what the problem behind ci failure is, will send v2 tomorrow.

raphaelsc commented:
v2: fix when() predicate, to return on shutdown

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10932/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10932/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3060/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11083/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10932/)
#### Build Details:
- Duration: 2 hr 4 min
- Builder: spider1.cloudius-systems.com

denesb commented:
> 🔴 CI State: FAILURE

CI failed due to unrelated test.py issue https://github.com/scylladb/scylladb/issues/20161.
All tests passed so the PR is good to go from this regard.

@raphaelsc please add backport labels and justification.

raphaelsc commented:
> > 🔴 CI State: FAILURE
> 
> CI failed due to unrelated test.py issue #20161. All tests passed so the PR is good to go from this regard.
> 
> @raphaelsc please add backport labels and justification.

done.

denesb commented:
@raphaelsc there are conflicts, please rebase.

If you want this to be backported, we need an issue for this, which the PR will fix.

raphaelsc commented:
> @raphaelsc there are conflicts, please rebase.
> 
> If you want this to be backported, we need an issue for this, which the PR will fix.

v3:
rebased
added FIXME tag for backport.

avikivity commented on service/storage_service.cc:
```diff
@@ -5278,7 +5278,7 @@ future<> storage_service::load_tablet_metadata(const locator::tablet_metadata_ch
 }
 
-future<> storage_service::process_tablet_split_candidate(table_id table) {
+future<> storage_service::process_tablet_split_candidate(table_id table) noexcept {
```
find_column_family() below can throw, and I'm sure split_all_storage_groups() can too. Slapping noexcept on a function isn't enough.

avikivity replied:
I suggest to keep it possibly throwing and follow  up with an exception safety fix (which won't be simple).

raphaelsc replied:
We handle exceptions and don't propagate it, in process_tablet_split_candidate(). All the calls outside the try catch block, are already noexcept. the reason I added noexcept is to document the behavior, as it's important that the split monitor keeps running despite any exception.

avikivity replied:
Ah, ok. Only the definition of the lambda calling find_column_family is outside the try block, not its execution.

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11188/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11188/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/237/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3282/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11307/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11188/)
#### Build Details:
- Duration: 2 hr 45 min
- Builder: spider3.cloudius-systems.com
